src/dataflow/src/source/postgres.rs

```diff
@@ -253,7 +259,12 @@ impl PostgresSourceReader {
 
             pin_mut!(reader);
             let mut mz_row = Row::default();
-            while let Some(b) = reader.next().await.transpose()? {
+            // TODO: once tokio-stream is released with https://github.com/tokio-rs/tokio/pull/4502
+            //    we can convert this into a single `timeout(...)` call on the reader CopyOutStream
+            while let Some(b) = tokio::time::timeout(Duration::from_secs(15), reader.next())
```

Comment:
15s is a bit of a guess for how long to wait for an individual row from `COPY OUT`, open to ideas here!

Reply:
Did you figure out why the TCP keepalive doesn't kick in while waiting on this `.next()` call?

Reply:
Please use 30 seconds for this, to be consistent with other timeouts we have in the system. This will become important when we test resilience with random tests that will cause random network disruptions but we never more than 30 seconds. The test will then expect that sub-30-second network disruptions will be survivable.

Reply:
> Did you figure out why the TCP keepalive doesn't kick in while waiting on this .next() call?

When I lowered it down, it actually did kick in. If Postgres was back up by the time the keepalive or one of its retries fired, Postgres would actually ACK (!) and the connection would stay up without error.

> Please use 30 seconds for this, to be consistent with other timeouts we have in the system. This will become important when we test resilience with random tests that will cause random network disruptions but we never more than 30 seconds. The test will then expect that sub-30-second network disruptions will be survivable.

Perfect, will update